src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -1,37 +1,53 @@
 import * as React from 'react';
 import AnnotatorContext from './AnnotatorContext';
-import { Action, AnnotationActionEvent, AnnotatorState, Status } from './types';
+import { Action, Annotator, AnnotationActionEvent, AnnotatorState, Status } from './types';
 
 export interface WithAnnotationsProps {
+    onAnnotator: (annotator: Annotator) => void;
     onAnnotatorEvent: ({ event, data }: { event: string; data: unknown }) => void;
     onPreviewDestroy: () => void;
 }
 
 export interface ComponentWithAnnotations {
+    emitActiveChangeEvent: (id: string | null) => void;
     getAction: (eventData: AnnotationActionEvent) => Action;
     handleActiveChange: (annotationId: string | null) => void;
+    handleAnnotationChangeEvent: (id: string | null) => void;
     handleAnnotationCreate: (eventData: AnnotationActionEvent) => void;
     handleAnnotatorEvent: ({ event, data }: { event: string; data?: unknown }) => void;
+    handleOnAnnotator: (annotator: Annotator) => void;
     handlePreviewDestroy: () => void;
 }
 
-export type WithAnnotationsComponent<P> = React.ComponentClass<P & WithAnnotationsProps>;
-
 const defaultState: AnnotatorState = {
     activeAnnotationId: null,
     annotation: null,
     action: null,
     error: null,
 };
 
+export type WithAnnotationsComponent<P> = React.ComponentClass<P & WithAnnotationsProps>;
+
 export default function withAnnotations<P extends object>(
     WrappedComponent: React.ComponentType<P>,
 ): WithAnnotationsComponent<P> {
     class ComponentWithAnnotations extends React.Component<P & WithAnnotationsProps, AnnotatorState> {
         static displayName: string;
 
+        annotator: Annotator | undefined;
+
         state = defaultState;
 
+        emitActiveChangeEvent = (id: string | null) => {
+            const { annotator } = this;
+
+            if (!annotator || !annotator.emit) {
```

Comment:
Is this an error condition? Should we handle it somehow other than swallowing it?

Reply:
No, the annotator is loaded through an event so it starts null or undefined, TS throws an error if you don't have a gate for it.